### PR TITLE
Update faq.md

### PR DESCRIPTION
**Pull request by Dref360**
## Summary:

First pass at trying to answer some common questions.
Let me know if this is useful and which questions should we add?

## Checklist:

* [x] Your code is documented (To validate this, add your module to `tests/documentation_test.py`).
* [x] Your code is tested with unit tests.
* [x] You moved your Issue to the PR state.

### CODE REVIEW

File: docs/faq.md [hunk 1]
```diff
@@ -109,6 +120,29 @@ labels = query_human(ranks, pool)
 active_dataset.label(ranks, labels)
 ```
 
+## Theory FAQ
+
+Bayesian active learning is a relatively small field will a lot of unknowns. This sections aims at presenting some of our findings so that newcomers can quickly learn.
```
Comment (parmidaatg) [inline docs/faq.md:125]:
```suggestion
Bayesian active learning is a relatively small field with a lot of unknowns. This section aims at presenting some of our findings so that newcomers can quickly learn.
```

File: docs/faq.md [hunk 2]
```diff
@@ -109,6 +120,29 @@ labels = query_human(ranks, pool)
...
+
+Don't forget to look at our [literature review](../literature/index.md) for a good introduction to the field.
+
+### Should you use early stopping?
+
+From our experiments, **early stopping hurts the process**. The training dataset is so small that the model overfits very quickly and hence early stopping triggers too early. We also know from [Atighehchian et al.](https://arxiv.org/abs/2006.09916) that **underfitting hurts the process more than overfitting**. 
+
+### Which optimizer works best?
+
+We find that **SGD works well in active learning**. More complex optimizers such as Adam hurt the process. This is mostly the case in the beginning of the process where the model overfits quickly because the training set is small.
```
Comment (parmidaatg) [inline docs/faq.md:135]: In the Intent classification, I use ADAM and since we reinitialize adam at every AL step it works really well. Let's add a citation to the `label efficiency` paper for the part that we indicate adam hurts since they mention that in their paper but also indicate that based on our experiments we could make AL work for distilled Bert with ADAM.

File: docs/faq.md [hunk 3]
```diff
@@ -109,6 +120,29 @@ labels = query_human(ranks, pool)
...
+
+### How do you evaluate active learning?
+
+The standard process is to compare to uniform sampling (sometime refered as *Random*). Some datasets are better to use than others. Academic datasets are often too clean for active learning because they were manually curated. 
+Remember that **active learning works best on industrial datasets** where duplicates, low-information examples or noisy examples are common.
+
+### Which query size to use?
+
+Of course the lower the better, but [Atighehchian et al.](https://arxiv.org/abs/2006.09916) shows that BALD works well with a query size under 1000. This was tested on an academic dataset where Random sampling is especially strong. In practice, BALD performs worse on low-diversity datasets and could wrongly behave on a lower query size.
+
```
Comment (parmidaatg) [inline docs/faq.md:146]: lets add a section on what to do if the test and train distribution are different (actually this can be a tutorial) wdyt?

File: docs/faq.md [hunk 4]
```diff
@@ -153,19 +189,23 @@ for al_step in range(NUM_AL_STEP):
...
 
-When using MC-Dropout, or any other Bayesian methods, you will want to compute the Bayesian model average (BMA) at test time too.
+When using MC-Dropout, or any other Bayesian methods, you will want to compute the Bayesian model average (BMA) at test
+time too.
 
-To do so, you can specify the `average_predictions` parameters in `ModelWrapper.test_on_dataset`. The prediction will be averaged over `iterations` stochastic predictions. 
+To do so, you can specify the `average_predictions` parameters in `ModelWrapper.test_on_dataset`. The prediction will be
+averaged over `iterations` stochastic predictions.
 
 This will slightly increase the ECE of your model and will improve the predictive performance as well.
```
Comment (parmidaatg) [inline docs/faq.md:204]: maybe we can add a note on recently available `unpatch` module?

**Review by parmidaatg — CHANGES_REQUESTED** [review]
amazing :)

File: docs/faq.md [hunk 5]
```diff
@@ -109,28 +130,65 @@ labels = query_human(ranks, pool)
...
+### Should you use early stopping?
+
+From our experiments, **early stopping hurts the process**. The training dataset is so small that the model overfits
+very quickly and hence early stopping triggers too early. We also know
+from [Atighehchian et al.](https://arxiv.org/abs/2006.09916) that **underfitting hurts the process more than
+overfitting**.
+
+### Which optimizer works best?
+
+We find that **SGD works well in for computer vision problems**. More complex optimizers such as Adam hurt the process.
```
Comment (Dref360) [inline docs/faq.md:149]:
```suggestion
We find that **SGD works well in for computer vision problems**. More complex optimizers such as Adam hurt the process. [Beck et al. 2021](https://arxiv.org/abs/2106.15324) find similar results.
```

File: docs/faq.md [hunk 6]
```diff
@@ -109,28 +130,65 @@ labels = query_human(ranks, pool)
...
+from [Atighehchian et al.](https://arxiv.org/abs/2006.09916) that **underfitting hurts the process more than
+overfitting**.
+
+### Which optimizer works best?
+
+We find that **SGD works well in for computer vision problems**. More complex optimizers such as Adam hurt the process. [Beck et al. 2021](https://arxiv.org/abs/2106.15324) find similar results.
+This is mostly the case in the beginning of the process where the model overfits quickly because the training set is
+small.
+
+When finetuning Transformers, we find that the Adam optimizer works well.
```
Comment (parmidaatg) [inline docs/faq.md:153]:
```suggestion
When finetuning Transformers, we find that the Adam optimizer works well if it is re-initialized at the beginning of each active learning step.
```

File: docs/faq.md (hunk 4 above)
Comment (Dref360) [reply docs/faq.md:226]: Done above

Comment (Dref360) [reply docs/faq.md:226]: Added above
